statemachine/errors.go

```diff
@@ -14,8 +14,23 @@ var ExceededErrorRate = errors.New("exceeded error rate")
 
 // Err represents an error that occurred while a stateful handler was running.
 type Err struct {
-	Time time.Time `json:"timestamp"`
-	Err  string    `json:"error"`
+	Time    time.Time `json:"timestamp"`
+	Err     string    `json:"error"`
```

Comment:
[nit] may be change the name, seems little confusing as Err refers either struct or string.

Reply:
yeah i agree... i don't think we can touch that field because it's already exposed. ideally i would just replace `Err` with an error field, but i think we need to preserve the existing interface

Reply:
I like this change. We should comment this struct better and explain why it is written this way. Comments should provide good historical background.

Reply:
good call, added/updated the comments to try and be more clear about the context/history